MantidQt/CustomInterfaces/inc/MantidQtCustomInterfaces/Reflectometry/ReflCommandBase.h

```diff
@@ -37,7 +35,7 @@ Code Documentation is available at: <http://doxygen.mantidproject.org>
 class ReflCommandBase : public ReflCommand {
```

Comment:
There is a weakness in our design here. It has allowed us to use our **common-code** implementation in place of our contract (hence the `ReflCommandBase` being passed around instead of the `ReflCommand`). Here are some suggestions:

1. Use `private` rather than public inheritance 
2. Make the base class a CRTP. This interesting pattern would prevent you from using the base-class ever, because the base class is really a templated type, which requires a type argument.
4. Do not inherit from `ReflCommandBase`, but forward all calls to the `ReflCommandBase` from your concrete types.